{{ cookiecutter.project_slug }}/Makefile

```diff
@@ -1,75 +1,7 @@
-.PHONY: .tf_version
+$(shell if [ -z "${ENV}" ]; then ENV = "dev"; fi)
 
-ifneq ($(CI), true)
-include .env
-endif
+init:
+	terraform -chdir=tf init -backend-config=$(ENV).tfbackend -reconfigure
```

Comment:
I'm not sure we can ditch docker-compose all together yet. They are folks running Mac/Arm64, windows,  Linux. The current architecture implements pattern called the 3 musketeers. I'll advise to have a look https://3musketeers.io/ it will give you more context I think.

Reply:
Am familiar with 3 musketeers
Does anyone have trouble running terraform itself? I think running docker compose here just adds extra steps/complexity without much benefit (and isn't even used in the pipeline which defeats the purpose it's used for in 3 musketeers)
Some people have also reported having issues with docker running on the M1 chips on the newer macs
If anyone can't run terraform itself it could be worth using docker, but I don't think we need compose to do that either way

Reply:
@daniellohrey-hai you've mentioned a few time how docker-compose adds complexity?  Can you please explain how?  I have to admit I don't see how it does.

A hypothetical example:
```
docker compose run --rm terraform apply
```
vs
```
docker run -v ${PWD}:/app -e AWS_REGION=ap-southeast-2 -e ENVIRONMENT=prod terraform apply
```

Also bear in mind Terraform will always be run locally and in CI, so the repo needs to cater for both, as it currently does.

I've heard of folks having issues with M1 and Docker with some of the more exotic libraries we use, but not with any IaC tooling like Terraform.  Can you let me know who in the Org is having issues with this?

I'm not equating docker compose to 3 musketeers.  Happy to ditch the latter if we feel it's no longer needed.

Reply:
Haven't looked too deeply into the M1 issue, but fair enough.

Currently running plan locally looks like:
```
plan: .env
	docker-compose run --rm envvars ensure --tags terraform
	docker-compose run --rm --workdir /app/$(DIR) terraform terraform plan
```
which is fine, the Makefile abstracts the length of of any commands.

My issue is with `envvars.yml` and `dockercompose.yml` being extra files, and steps that can go wrong (that are difficult to debug). Seems unnecessary to me, but if you think it's needed then happy to leave it.

Is the plan to run docker in CI as well?
And why is running docker locally necessary? have people reported issues running terraform?

Reply:
I'm not actually super concerned where we end up with a lot of this, as long as we make a thought-out decision and stick with it; my main concern is having things in a state where we can start to work on some reusable automation (across Harrison at least).

Reply:
> And why is running docker locally necessary? have people reported issues running terraform?

There are many reasons to run Terraform locally as well as in CI.  We pin to specific versions of Terraform.  With a number of repos in existence, this isn't the same across all repos.  Juggling these versions outside of docker?  No thanks.

Three people (myself, Daniel C and Issa) have responded to you that we are not keen to ditch Docker and compose.

Reply:
Nice, that reason makes sense, thanks for calling that out. Not sure if `There are many reasons to run Terraform locally as well as in CI.` is a typo for docker, but would like to clarify; is the plan to run terraform with docker in CI as well, or just locally?

What are you thoughts on the other proposed changes? Keen to get some automation going across Harrison.

Reply:
I have responded here - https://github.com/harrison-ai/cookiecutter-terraform/pull/21#issuecomment-1350776798